EduID/SharedViews/TwoFactorKeyControlCollapsible.swift

```diff
@@ -0,0 +1,213 @@
+                font: .sourceSansProSemiBold(size: 14),
+                color: .secondaryColor,
+                lineSpacing: 6)
+        )
+        
+        let uniqueIdStack = UIStackView(arrangedSubviews: [uniqueIdLabel, uniqueIdValue])
+        uniqueIdStack.axis = .horizontal
+        uniqueIdStack.distribution = .fillEqually
+        
+        // line 4
+        let line4 = UIView()
+        line4.height(1)
+        line4.backgroundColor = .backgroundColor
+        
+        // Biometrics
+        let biometricsLabel = UILabel()
+        biometricsLabel.attributedText = NSAttributedString(
+            string: L.TwoFactorKeys.Label.UseBiometrics.localization,
+            attributes: AttributedStringHelper.attributes(
+                font: .sourceSansProRegular(size: 14),
+                color: .secondaryColor,
+                lineSpacing: 6)
+        )
+        
+        let biometricsValue = UISwitch()
+        biometricsValue.tintColor = .backgroundColor
+        biometricsValue.addTarget(self, action: #selector(biometricsSwitchToggled), for: .valueChanged)
+        biometricsValue.isOn = identity.biometricIDEnabled == NSNumber(true)
+        biometricsValue.isEnabled = ServiceContainer.sharedInstance().secretService.biometricIDAvailable
+        
+        let biometricsStack = UIStackView(arrangedSubviews: [biometricsLabel, biometricsValue])
+        uniqueIdStack.axis = .horizontal
+        uniqueIdStack.distribution = .fillEqually
+        
+        // line 5
+        let line5 = UIView()
+        line5.height(1)
+        line5.backgroundColor = .backgroundColor
+        
+        // remove button
+        let button = EduIDButton(type: .borderedRed, buttonTitle: L.TwoFactorKeys.DeleteKey.localization)
+        button.addTarget(self, action: #selector(buttonAction), for: .touchUpInside)
+        
+        stack = UIStackView(arrangedSubviews: [
+            bodyStack,
+            line2,
+            firstLoginStack,
+            line3,
+            uniqueIdStack,
+            line4,
+            biometricsStack,
+            line5,
+            button
+        ])
+        stack.axis = .vertical
+        stack.distribution = .equalSpacing
+        stack.spacing = 18
+        
+        addSubview(stack)
+        stack.edges(to: self, insets: TinyEdgeInsets(top: 12, left: 18, bottom: 12, right: 18))
+        
+        //border
+        layer.borderWidth = 3
+        layer.borderColor = UIColor.backgroundColor.cgColor
+        layer.cornerRadius = 6
+        
+        // initially hide elements
+        for i in (1..<stack.arrangedSubviews.count) {
+            stack.arrangedSubviews[i].isHidden = true
+            stack.arrangedSubviews[i].alpha = 0
+        }
+    }
+    
+    //MARK: - expand or contract action
+    @objc
+    func expandOrContract() {
+        UIView.animate(withDuration: 0.3) { [weak self] in
+            for i in (1..<(self?.stack.arrangedSubviews.count ?? 0)) {
```

Comment:
i’ve always used ‘guard let self else { return }’ in this project, that would make things more consistent.

Reply:
I also do when there are multiple self references in a block. I prefer writing one `?` vs the long guard text, but when you also have to do nil checks then yes it is just easier to do the guard check in the beginning for the block.
PS. Swift should have a shorthand for this, I wrote that line so many times... :)